torch/csrc/jit/custom_operator.h

```diff
@@ -3,218 +3,9 @@
-      "argument",
-      inferredSchema.arguments(),
-      providedSchema.arguments(),
-      inferredSchema,
-      providedSchema);
-  checkArgumentVector(
-      "return value",
-      inferredSchema.returns(),
-      providedSchema.returns(),
-      inferredSchema,
-      providedSchema);
-  return providedSchema;
-}
-} // namespace detail
-
-/// Registers a custom operator with a name or schema, and an implementation
-/// function.
-///
-/// If the first argument specifies only the function name like `foo::bar`, the
-/// schema, including the type of each argument and the return type, is inferred
-/// from the function signature. Otherwise, the string should specify the whole
-/// schema, like `foo::bar(Tensor a, double b) -> Tensor`. In that case, the
-/// schema will still be inferred from the function and checked against this
-/// provided schema.
-///
-/// If the schema is left to be inferred, the argument names will take on
-/// sequential placeholder names like `_0`, `_1`, '_2' and so on. If you want
-/// argument names to be preserved, you should provide the schema yourself.
-///
-/// The implementation function can be a function pointer or a functor
-/// (including a lambda object). The function (or `operator()`) can take any
-/// number of arguments with a type from the subset accepted by the PyTorch
-/// JIT/Script backend, and return a single type or a tuple of types.
-///
-/// Example invocation:
-/// ```
-/// createOperator(
-///    "foo::bar(float a, Tensor b)",
-///    [](float a, at::Tensor b) { return a + b; });
-/// ```
-template <typename Implementation>
-Operator createOperator(
-    const std::string& schemaOrName,
-    Implementation&& implementation) {
-  using Traits = c10::guts::infer_function_traits_t<Implementation>;
-  using ArgumentTypes =
-      c10::guts::typelist::map_t<decay_t, typename Traits::parameter_types>;
-  using ArgumentTuple =
-      typename c10::guts::typelist::to_tuple<ArgumentTypes>::type;
-  static constexpr auto kNumberOfArguments =
-      std::tuple_size<ArgumentTuple>::value;
-
-  auto schema = torch::jit::detail::inferAndCheckSchema<Traits>(schemaOrName);
-
-  // [custom operator aliasing] Currently, we have no way for the user to
-  // specify the alias annotations for a custom op. Therefore, we have to:
-  //   1. Assume that custom ops will mutate all inputs, have side effects, and
-  //      produce wildcard outputs.
-  //   2. Have some way of distinguishing between a custom op and a builtin op
-  //      so that we can apply the above rule.
-  // We do this by manually whitelisting "aten" and "prim" namespaces as
-  // builtins.
-  //
-  // We don't want to preserve this distinction between custom/builtin ops, as
-  // it is fragile and hard to maintain. When we provide a way for op
-  // registration to specify alias annotations, we should fix up builtins to
-  // use that and remove all references to this note.
-  Symbol name = Symbol::fromQualString(schema.name());
```

Comment:
that was kind of a good check, but I'm not sure where to plug it in the new version. Is there a public only API wrapper?

Reply:
no, we intentionally use the same API internally and externally. Maybe it's time to actually do what the comment is suggesting.